source/nodejs/adaptivecards/src/card-elements.ts

```diff
@@ -6233,6 +6237,18 @@ export class Container extends ContainerBase {
         return result;
     }
 
+    getCurrentCarouselId(): string | undefined {
+        if (this._swiper) {
+            if (this._swiper.slides.length) { 
+                let activeSlide = this._swiper.slides[this._swiper.activeIndex] as HTMLElement;
+                if (activeSlide) { 
+                    return activeSlide.id;
+                }
+            }
+        }
+        return undefined;
+    }
```

Comment:
could be a little tighter:

```suggestion
    getCurrentCarouselId(): string | undefined {
        if (this._swiper?.slides?.length) {
            const activeSlide = this._swiper.slides[this._swiper.activeIndex] as HTMLElement;
            return activeSlide?.id;
        }
        return undefined;
    }
```